crates/ruff_linter/src/rules/pep8_naming/rules/invalid_function_name.rs

```diff
@@ -84,6 +82,43 @@ pub(crate) fn invalid_function_name(
         return;
     }
 
+    let parent_class = semantic
+        .current_statement_parent()
+        .and_then(|parent| parent.as_class_def_stmt());
+
+    // Ignore the visit_* methods of the ast.NodeVisitor and ast.NodeTransformer classes.
+    // Only applies if the Python version is less than 3.12.
+    // If Python is greater than 3.12, typing.override should be used instead.
+
+    let is_ast_visitor = checker.target_version() < PythonVersion::PY312
```

Comment:
I think we can remove this check. I'm not a fan of the change itself but it's also sort of fine. 

If we keep this check, then it should apply for all python versions unless typing_extensions is disallowed by https://docs.astral.sh/ruff/settings/#lint_typing-extensions